services/QuillLMS/app/services/diagnostics_organized_by_classroom_fetcher.rb

```diff
@@ -28,46 +30,27 @@ def run
       index_of_existing_classroom = classrooms.find_index { |c| c['id'] == record['classroom_id'] }
       name = grouped_name(record)
 
-      next if record['post_test_id'] &&
-              index_of_existing_classroom &&
+      next if index_of_existing_classroom &&
               classrooms[index_of_existing_classroom]['diagnostics'].find { |diagnostic| diagnostic[:name] == name }
 
       grouped_record = {
         name: name,
-        pre: record
+        pre: record_with_aggregated_activity_sessions(
+          record['activity_id'],
+          record['classroom_id']
+        )
       }
 
       if record['post_test_id']
         post_test = record_with_aggregated_activity_sessions(
           record['post_test_id'],
-          record['classroom_id'],
-          record['activity_id']
+          record['classroom_id']
         )
 
         grouped_record[:post] = post_test || {
           activity_name: Activity.find_by_id(record['post_test_id'])&.name,
           unit_template_id: ActivitiesUnitTemplate.find_by_activity_id(record['post_test_id'])&.unit_template_id
         }
-
-        grouped_record[:pre] = record_with_aggregated_activity_sessions(
-          record['activity_id'],
-          record['classroom_id'],
-          nil
-        )
-      else
```

Comment:
Similarly, here, I think this special handling of pre-tests that don't have post-tests is no longer necessary, so I cleaned up the code.